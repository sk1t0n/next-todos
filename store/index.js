import { configureStore } from '@reduxjs/toolkit';
import todoReducer from './todoSlice';
import menuReducer from './menuSlice';

export default configureStore({
  reducer: {
    todos: todoReducer,
    menu: menuReducer
  }
});
