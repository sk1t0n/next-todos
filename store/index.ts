import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import todoReducer from './slices/todoSlice';
import menuReducer from './slices/menuSlice';
import userReducer from './slices/userSlice';

export const rootReducer = combineReducers({
  todos: todoReducer,
  menu: menuReducer,
  user: userReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer
});

export const persistor = persistStore(store);
export default store;
