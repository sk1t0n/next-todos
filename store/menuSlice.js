import { createSlice } from '@reduxjs/toolkit';

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    current: 'home'
  },
  reducers: {
    setCurrent: (state, action) => {
      state.current = action.payload.current;
    }
  }
});

export const { setCurrent } = menuSlice.actions;

export default menuSlice.reducer;
