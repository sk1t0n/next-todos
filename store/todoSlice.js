import { createSlice } from '@reduxjs/toolkit';

const todoSlice = createSlice({
  name: 'todos',
  initialState: {
    todos: []
  },
  reducers: {
    addTodo: (state, action) => {
      state.todos.push({
        id: Date.now(),
        text: action.payload.text,
        completed: false
      });
    },
    updateTodo: (state, action) => {
      state.todos = state.todos.map(todo => {
        if (todo.id !== action.payload.id) return todo;

        return {
          ...todo,
          completed: !todo.completed
        };
      });
    },
    removeTodo: (state, action) => {
      state.todos = state.todos.filter(
        todo => todo.id !== action.payload.id
      );
    }
  }
});

export const { addTodo, updateTodo, removeTodo } = todoSlice.actions;

export default todoSlice.reducer;
