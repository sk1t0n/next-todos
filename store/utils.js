import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import menuReducer from './menuSlice';
import todoReducer from './todoSlice';

export const renderWithRedux = (
  component,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        menu: menuReducer,
        todos: todoReducer
      },
      preloadedState
    }),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return render(component, { wrapper: Wrapper, ...renderOptions })
};
