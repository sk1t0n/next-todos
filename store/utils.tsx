import React from 'react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { RenderOptions } from '@testing-library/react/types';
import { EnhancedStore } from '@reduxjs/toolkit/dist/configureStore';
import { PreloadedState } from '@reduxjs/toolkit';
import { RootState, rootReducer } from './index';

type Config = {
  preloadedState?: PreloadedState<RootState>;
  store?: EnhancedStore;
  renderOptions?: Omit<RenderOptions, 'queries'>;
};

export const renderWithRedux = (
  component: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      reducer: rootReducer,
      preloadedState
    }),
    ...renderOptions
  }: Config = {}
) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return render(component, { wrapper: Wrapper, ...renderOptions })
};
