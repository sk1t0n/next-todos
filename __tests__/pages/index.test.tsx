import { screen } from '@testing-library/react';
import { renderWithRedux } from '../../store/utils';
import { RenderResult } from '@testing-library/react/types';
import Home from '../../pages/index'

describe('Home', () => {
  it('check the snapshot', () => {
    const component = <Home />;
    const {asFragment} = renderWithRedux(component);
    expect(asFragment()).toMatchSnapshot();
  });
});
