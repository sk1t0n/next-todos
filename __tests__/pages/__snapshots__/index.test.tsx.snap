// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home check the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-xs-22 ant-col-xs-offset-1 ant-col-sm-20 ant-col-sm-offset-2 ant-col-md-18 ant-col-md-offset-3 ant-col-lg-12 ant-col-lg-offset-6"
    >
      <div
        class="ant-spin ant-spin-lg ant-spin-spinning"
        style="display: flex; justify-content: center; margin-top: 25vh;"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
